<a href="https://colab.research.google.com/github/yumeean/CS482/blob/main/Homework1_ig62.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Homework 1
#Ian Gabrielle Gojo Cruz
#CS 482

##1)  A data scientist develops a model of the mortality probability distribution function  
<p align="center">
$p(t) = 3 \times 10^{-9} t^2(100-t)^2, 0 \le t \le 100 ~\text{years}$  
</p>

$p(t)$ is 0.0 outside the above range of t.  

###a. What is the probability that a person will die between 60 and 70.  
  
Given : $p(t) = 3 \;x \;10^{-9}t^{2}(100 - t)^{2}$, for $0 \le t \le 100$ years  

Find  : $P(60 \le t \le 70)$

$\int_{60}^{70}3 \;x \;10^{-9}t^{2}(100 - t)^{2} \;dt$  

In [ ]:
from scipy.integrate import quad

def integrand(t):
  return 3*10**(-9)*t**2*(100 - t)**2

ans, err = quad(integrand, 60, 70)
print(ans)

0.15436000000000002


<font color='blue'>Answer:  $\int_{60}^{70}3 \;x \;10^{-9}t^{2}(100 - t)^{2} \;dt$  = $0.15436$

---



###b. What is the probability that a person will die between 60 and 70, given that was alive at 60.  
  
Find: $P(60 \le t \le 70 | 1 - P(0 \le t \le 60))$  

Finding the probability that the person is alive at age 60 would be:

Probability that the person will die from $ 0 \;to \;60 $  
  
$P(0 \le t \le 60) = \int_{0}^{60}3 \;x \;10^{-9}t^{2}(100 - t)^{2} \;dt$

In [ ]:
ans, err = quad(integrand, 0, 60)
print(ans)

0.6825600000000001


$P(0 \le t \le 60) = 0.68256$  
  
Meaning, the probability that the person is alive at age 60 would be 
   
$ 1 - 0.68256 = 0.31744$
  
Conditional Probability:  

$P(B|A) = \frac{P(A \;\cap \;B)}{P(A)}$

We have the following:  
let &nbsp;$P(B) = P(60 \le t \le 70) = 0.15436$  
&nbsp; &nbsp; &nbsp; $P(A) = 1 - P(0 \le t \le 60) = 0.31744$
  
then,  

$P(B|A) = \frac{0.15436 \;x \;0.31744}{0.31744} = 0.15436$  
  
<font color='blue'>Answer: $P(60 \le t \le 70 | 1 - P(0 \le t \le 60)) = 0.15436$ 


##2) Three switches connected in parallel operate independently. Each switch remains closed with probability p.

![figure 1](https://pantelis.github.io/cs634/docs/common/assignments/probability-assignment/images/probabilistic-switches.png)  

### a. Find the probability of receiving an input signal at the output.  
  
Given: $P(S_1)=P(S_2)=P(S_3)=p$ (switched is closed)  
  
$ 1 - p $ (switch is open)
  
Let A = $S_i$ is closed  
  
Let B = Receiving an input signal at the output 
   
We know that that $S_1, S_2,$ and $S_3$ are all independent meaning:  
<p align="center">
$P(A_1A_2A_3) = P(A_1)P(A_2)P(A_3)$   
</p

In order for event B to happen, at least 1 of 3 switches should be closed.  
<p align="center">
$B = A_1 \cup A_2 \cup A_3$
</p>
$P(B) = P(A_1 \cup A_2 \cup A_3)$  

$P(B) = 1 - (1-p)^3$  

<font color='blue'>Answer: $P(B) = 3p - 3p^2 + p^3$

### b. Find the probability that switch $S_i$ is open given that an input signal is received at the output.  

Given the information above, we need to find  

$P(\overline{A_1}\;|\;B) = \;?$  

Using Baye's Theorem,  where $P(A\;|\;B) = \frac{P(B|A)\;P(A)}{P(B)}$  then,  


$P(\overline{A_1}\;|\;B) = \frac{P(B|\overline{A_1})\;P(\overline{A_1})}{P(B)}$  

We also know that $P(B|A_1) = 1$, and  $P(B|\overline{A_1}) = P(A_2 \cup A_3) = 2p - p^2$  

We can just fill these values in our equation resulting to

<font color='blue'>Answer: $P(\overline{A_1}|B) = \frac{(2p - p^2)(1 - p)}{3p - 3p^2 + p^3}$